test/e2e/framework.go

```diff
@@ -1429,30 +1439,36 @@ func (data *TestData) deployFlowAggregator() error {
 		return fmt.Errorf("error when deploying the Flow Aggregator; %s not available on the control-plane Node", flowAggYaml)
 	}
-
-	if rc, _, _, err = data.provider.RunCommandOnNode(controlPlaneNodeName(), fmt.Sprintf("kubectl -n %s rollout status deployment/%s --timeout=%v", flowAggregatorNamespace, flowAggregatorDeployment, 2*defaultTimeout)); err != nil || rc != 0 {
-		_, stdout, _, _ := data.provider.RunCommandOnNode(controlPlaneNodeName(), fmt.Sprintf("kubectl -n %s describe pod", flowAggregatorNamespace))
-		_, logStdout, _, _ := data.provider.RunCommandOnNode(controlPlaneNodeName(), fmt.Sprintf("kubectl -n %s logs -l app=flow-aggregator", flowAggregatorNamespace))
-		return fmt.Errorf("error when waiting for the Flow Aggregator rollout to complete. kubectl describe output: %s, logs: %s", stdout, logStdout)
-	}
-	// Check for flow-aggregator pod running again for db connection establishment
-	flowAggPod, err := data.GetFlowAggregator()
+	// Check for flow-aggregator pod running and for db connection establishment.
+	flowAggPodName, err := data.GetFlowAggregator()
 	if err != nil {
 		return fmt.Errorf("error when getting flow-aggregator Pod: %v", err)
 	}
-	podName := flowAggPod.Name
-	_, err = data.PodWaitFor(defaultTimeout*2, podName, flowAggregatorNamespace, func(p *corev1.Pod) (bool, error) {
-		for _, condition := range p.Status.Conditions {
-			if condition.Type == corev1.PodReady {
-				return condition.Status == corev1.ConditionTrue, nil
-			}
-		}
-		return false, nil
-	})
+	err = data.PodWaitForReady(2*defaultTimeout, flowAggPodName, flowAggregatorNamespace)
 	if err != nil {
-		_, stdout, stderr, podErr := data.provider.RunCommandOnNode(controlPlaneNodeName(), fmt.Sprintf("kubectl get pod %s -n %s -o yaml", podName, flowAggregatorNamespace))
+		_, stdout, stderr, podErr := data.provider.RunCommandOnNode(controlPlaneNodeName(), fmt.Sprintf("kubectl get pod %s -n %s -o yaml", flowAggPodName, flowAggregatorNamespace))
 		return fmt.Errorf("error when waiting for flow-aggregator Ready: %v; stdout %s, stderr: %s, %v", err, stdout, stderr, podErr)
 	}
+	var errLog error
+	var logString string
+	// Check log of flow-aggregator pod to make sure ClickHouse is enabled and connected to flow-aggregator
+	if err := wait.Poll(defaultInterval, defaultTimeout, func() (bool, error) {
+		logString, err = data.GetPodLogs(flowAggregatorNamespace, flowAggPodName, &corev1.PodLogOptions{})
+		if err != nil {
+			errLog = fmt.Errorf("error when getting Flow Aggregaotr logs: %v", err)
+			return false, nil
+		}
+		if strings.Contains(logString, "error when creating ClickHouse export process") {
+			errLog = fmt.Errorf("error when creating ClickHouse export process")
+			return false, nil
+		}
+		if strings.Contains(logString, "Starting ClickHouse exporting process") {
+			return true, nil
+		}
+		return false, nil
+	}); err != nil {
+		return fmt.Errorf("error when checking log for Flow Aggregator: %v\nLog: %s", errLog, logString)
```

Comment:
when will errLog and logString be populated for this line ?

Reply:
They will be printed when the Flow Aggregator cannot connect to ClickHouse in defaultTimeout or when GetPodLogs returns errors.

Reply:
Maybe print the logString only when errLog is empty?

Reply:
I've removed printing the logString as I think it might be redundant to print the full log.